website/docs/reference/global-configs/legacy-behaviors.md

```diff
@@ -12,16 +12,77 @@ Another category of flags provides existing projects with a migration window for
+1. **Introduction (disabled by default):** dbt adds logic to support both 'old' + 'new' behaviors. The 'new' behavior is gated behind a flag, disabled by default, preserving the old behavior.
+2. **Maturity (enabled by default):** The default value of the flag is switched, from `false` to `true`, enabling the new behavior by default. Users can preserve the 'old' behavior and opt out of the 'new' behavior by setting the flag to `false` in their projects. They may see deprecation warnings when they do so.
+3. **Removal (generally enabled):** After marking the flag for deprecation, we remove it along with the 'old' behavior it supported from the dbt codebases. We aim to support most flags indefinitely, but we're not committed to supporting them forever. If we choose to remove a flag, we'll offer significant advance notice.
 
-These flags can _only_ be set in `dbt_project.yml`. They configure behaviors closely tied to project code. They should be defined in version control, and changed via pull/merge request, with the same testing and peer review.
+## Behavior change flags
 
-## Behaviors
+These flags _must_ be set in the `flags` dictionary in `dbt_project.yml`. They configure behaviors closely tied to project code, which means they should be defined in version control and modified through pull or merge requests, with the same testing and peer review.
+
+The following example displays the current flags and their current default values in the latest dbt Cloud and dbt Core versions. To opt out of a specific behavior change, set the values of the flag to `False` in `dbt_project.yml`. You'll continue to see warnings for legacy behaviors that you have opted out of explicitly until you either resolve them (switch the flag to `True`) or choose to silence the warnings.
+
+<File name='dbt_project.yml'>
+
+```yml
+flags:
+  require_explicit_package_overrides_for_builtin_materializations: False
+  require_model_names_without_spaces: False
+  source_freshness_run_project_hooks: False
+```
+
+</File>
+
+| Flag                                                            | dbt Cloud: Intro | dbt Cloud: Maturity | dbt Core: Intro | dbt Core: Maturity | 
+|-----------------------------------------------------------------|------------------|---------------------|-----------------|--------------------|
+| require_explicit_package_overrides_for_builtin_materializations | 2024.04.141      | 2024.05.xxx         | 1.6.14, 1.7.14    | 1.8.0             |
+| require_resource_names_without_spaces                           | 2024.05.xxx      | 2024.06.xxx         | 1.8.0           | 1.9.0             |
+| source_freshness_run_project_hooks                              | 2024.03.61       | 2024.06.xxx         | 1.8.0           | 1.9.0             |
+
+<sup>1</sup>dbt Cloud - "Keep on latest version"
+
+### `require_explicit_package_overrides_for_builtin_materializations`
```

Comment:
for the headers, i'm not sure theyre easy to read in the right toc. wondering if we can make the headers be shorter and more direct while including a brief description of the flag right within the header itself?
![Screenshot 2024-05-01 at 12 43 15](https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/46db934a-c0d1-4fd5-abc9-6cbc6268da51)

Reply:
if youre ok with it, i can update the links in the release notes and guides!

Reply:
Agree, they look awkward in the sidebar. Without headers is fine. We have links in `dbt-core` pointing to this page with anchors, it's no big deal if the anchors don't resolve correctly (the page load still works fine), I'll just update those links for v1.8 / "KoLV" once we've merged this PR.